harvester_e2e_tests/fixtures/rancher_api_client.py

```diff
@@ -20,28 +20,59 @@ def rancher_api_client(request):
     return api
 
 
+def _pickup_k8s_version(versions: [str], target_version: str):
+    """
+    versions:       e.g. ['v1.26.16-rancher2-3', 'v1.25.15-rancher1-1', ...]
+    target_version: e.g. v1.26, v1.26.16...
+    """
+    # e.g. v1.26.16-rancher2-3 / v1.26.16+rke2r1 will be sort as v1.26.16
+    sorted_vers = sorted(versions,
+                         key=lambda v: parse_version(v.split("+")[0].split("-")[0]),
+                         reverse=True)
+
+    for ver in sorted_vers:
+        if ver.startswith(target_version):
+            warnings.warn(UserWarning(f"Adopt {ver} for target k8s-version {target_version}"))
+            return ver
+
+    assert False, (
+        f"No supported version fits target k8s-version {target_version}\n"
+        f"Rancher endpoint supports {versions}"
+    )
+
+
+@pytest.fixture(scope='session')
+def rke1_version(request, rancher_api_client):
+    target_ver = request.config.getoption("--k8s-version")
+
+    code, data = rancher_api_client.settings.get("k8s-versions-current")
+    assert 200 == code, (code, data)
+    supported_vers = data["value"].split(",")
+    assert supported_vers
```

Comment:
There might have a potential bug the value is a empty string like:
```python3
>>> ''.split(",")
['']
```

then the assertion will never failed.
but consider that the following `_pickup_k8s_version` will also raise assertion error, so we can simply remove this one.